src/cpp/core/StringUtils.cpp

```diff
@@ -262,29 +262,36 @@ std::string utf8ToSystem(const std::string& str,
       return std::string();
 
 #ifdef _WIN32
+
    std::vector<wchar_t> wide(str.length() + 1);
-   int chars = ::MultiByteToWideChar(CP_UTF8, 0, str.c_str(), -1, &wide[0], static_cast<int>(wide.size()));
+   int chars = ::MultiByteToWideChar(
+            CP_UTF8, 0,
+            str.c_str(), -1,
+            &wide[0], static_cast<int>(wide.size()));
+
    if (chars < 0)
    {
       LOG_ERROR(LAST_SYSTEM_ERROR());
       return str;
    }
 
    std::ostringstream output;
-   char mbbuf[10];
-   // Only go up to chars - 1 because last char is \0
-   for (int i = 0; i < chars - 1; i++)
+   char buffer[16];
```

Comment:
Where did 16 come from? MSDN and C++ docs say `MB_CUR_MAX` or at worst `MB_LEN_MAX` are appropriate. These style of functions are generally scary; we weren't allowed to use them in MS-Office code at all, replaced with `wctomb_s`.